tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -52,17 +62,23 @@ namespace {
 class SPIRVEmitter : public ASTConsumer {
 public:
   explicit SPIRVEmitter(CompilerInstance &ci)
-      : theCompilerInstance(ci), outStream(*ci.getOutStream()), theContext(),
-        theBuilder(&theContext) {}
+      : theCompilerInstance(ci), outStream(*ci.getOutStream()),
+        diags(ci.getDiagnostics()), theContext(), theBuilder(&theContext) {}
 
   void AddRequiredCapabilitiesForExecutionModel(spv::ExecutionModel em) {
     if (em == spv::ExecutionModel::TessellationControl ||
         em == spv::ExecutionModel::TessellationEvaluation) {
       theBuilder.requireCapability(spv::Capability::Tessellation);
-      assert(false && "Tasselation Shaders are currently not supported.");
+      auto diagId = diags.getCustomDiagID(
+          clang::DiagnosticsEngine::Error,
```

Comment:
I'd suggest to create wrapper methods for diagnostics since the same code sequence will appear frequently:
```c++
DiagnosticBuilder error(const char* message) {
  auto diagId = diags.getCustomDiagId(..., message);
  return diags.Report(diagId);
}
```

similarly for `warning()`.